planner/core/task.go

```diff
@@ -2026,11 +2026,15 @@ func (t *mppTask) convertToRootTaskImpl(ctx sessionctx.Context) *rootTask {
 		StoreType: kv.TiFlash,
 	}.Init(ctx, t.p.SelectBlockOffset())
 	p.stats = t.p.statsInfo()
+
+	p.cost = t.cst / p.ctx.GetSessionVars().CopTiFlashConcurrencyFactor
```

Comment:
Maybe we should check if `CopTiFlashConcurrencyFactor` is 0?

Reply:
I tested and found that if `tidb_opt_tiflash_concurrency_factor` is set to 0, `p.cost` will be `+inf`, which seems a little reasonable……

I will limit the value range of this variable so that it can't be less than 1, cc @hanfei1991

Reply:
setting it to zero will not compare the costs of different mpp plans...

Reply:
we can do some meddling during comparing the cost: If the task is oriented from mpp Task, it will be superior to the non-mpp ones.

Reply:
> we can do some meddling during comparing the cost: If the task is oriented from mpp Task, it will be superior to the non-mpp ones.

I think this pr is mainly for user interface and a simple but reliable implement. How about merging this first and then treat "the estCost of mpp task will be 0 when tidb_allow_mpp is set to `enforce`" as an issue?

Reply:
> Maybe we should check if `CopTiFlashConcurrencyFactor` is 0?

Yes, now this restriction is added in the variable framework and is tested.